spec/lib/msf/core/exploit/remote/remote/http/http_cookie_spec.rb

```diff
@@ -0,0 +1,815 @@
+        invalid_str = random_string
+
+        expect do
+          cookie.max_age = invalid_str
+        end.to raise_error(ArgumentError)
+      end
+    end
+
+    describe 'Complex Object' do
+      it 'throws an TypeError with a Complex Object that cannot be converted into an Integer' do
+        obj = [rand(9), rand(9), rand(9)]
+
+        expect do
+          cookie.max_age = obj
+        end.to raise_error(TypeError)
+      end
+    end
+
+    describe 'nil' do
+      it 'assigns a nil to max_age successfully' do
+        cookie.max_age = nil
+        expect(cookie.max_age).to eql(nil)
+      end
+    end
+  end
+
+  describe 'expires' do
+    describe 'Time' do
+      it 'instance is assigned to expires successfully' do
+        t = Time.now
+
+        cookie.expires = t
+
+        expect(cookie.expires).to eql(t)
+        expect(cookie.expires.class).to eql(Time)
+      end
+    end
+
+    describe 'Non-Time' do
+      it 'which can be parsed to a Time object, will be assigned to expires successfully' do
+        obj = Time.now.to_s
+
+        cookie.expires = obj
+
+        expect(cookie.expires).to eql(Time.parse(obj))
+      end
+
+      it 'which cannot be parsed to a Time object, when assigned to expires, will throw a TypeError' do
+        obj = [rand(9), rand(9), rand(9)]
+
+        expect do
+          cookie.expires = obj
+        end.to raise_error(TypeError)
+      end
+    end
+
+    describe 'nil' do
+      it 'assigned to expires successfully' do
+        cookie.expires = nil
+        expect(cookie.expires).to eql(nil)
+      end
+    end
+  end
+
+  describe 'path' do
+    describe 'String' do
+      it 'beginning with "/" is passed to "path" successfully' do
+        p = '/' + random_string
+
+        cookie.path = p
+
+        expect(cookie.path).to eql(p)
+        expect(cookie.path.class).to eql(String)
+      end
+
+      it 'not beginning with "/" is passed to "path" and is set as "/"' do
+        p = random_string
+
+        cookie.path = p
+
+        expect(cookie.path).to eql('/')
+        expect(cookie.path.class).to eql(String)
+      end
+
+      it 'that is empty is passed to "path" and is set as "/"' do
+        p = ''
+
+        cookie.path = p
+
+        expect(cookie.path).to eql('/')
+        expect(cookie.path.class).to eql(String)
+      end
+    end
+
+    # If the Object A responds to to_str with a truthy Object that responds true to start_with?('/'),
+    # then path is set to A
+    describe 'Complex Object' do
+      it 'which is a kind of String will be passed to path successfully' do
+        class MyString < String
+        end
+        str = "/#{random_string(0)}"
+        my_str = MyString.new(str)
+
+        cookie.path = my_str
+
+        expect(cookie.path).to eql(str)
+      end
+    end
+
+    describe 'nil' do
+      it 'assigned to path successfully' do
+        n = nil
+
+        cookie.expires = n
+
+        expect(cookie.expires).to eql(n)
+      end
```

Comment:
"assigned to path" but you're testing expires

Reply:
👍